src/HermesTransport.InMemory/InMemoryMessageBroker.cs

```diff
@@ -0,0 +1,145 @@
+using HermesTransport;
+using System.Collections.Concurrent;
+using System.Threading.Channels;
+
+namespace HermesTransport.InMemory;
+
+public class InMemoryMessageBroker : IMessageBroker, IDisposable
```

Comment:
It would be nice to have Extensions file, that would register all necessary implementations in IServiceCollection for DI usage. 
Also this broker or subscribers could be registered as HostedService,  that would actively check if new messages are added to the channel.